diff -bur src.orig/config/site.mcr.proto src/config/site.mcr.proto
--- src.orig/config/site.mcr.proto	2005-04-29 01:30:47.000000000 +0200
+++ src/config/site.mcr.proto	2006-04-05 13:57:10.000000000 +0200
@@ -89,6 +89,13 @@
  */
 LINUX_PATCH_DEFINES = \
 /*  -DMMAP_LINUX_PATCH  */
+/*  -DXATTR_LINUX24_PATCH */
+/*  XATTR_LINUX24_PATCH
+ *     define this if your kernel has the setxattr()/getxattr()
+ *     syscalls; present in some kernels of the 2.4 series 
+ *     (comes with the POSIX ACL patch).
+ */
+
 
 /*****************************************************/
 /* Linux portability layer defines end.              */
diff -bur src.orig/gpl-linux/acl.c src/gpl-linux/acl.c
--- src.orig/gpl-linux/acl.c	2005-04-29 01:35:17.000000000 +0200
+++ src/gpl-linux/acl.c	2006-04-05 14:02:21.000000000 +0200
@@ -32,8 +32,7 @@
  *************************************************************************** */
 /* @(#)00       1.18  src/avs/fs/mmfs/ts/kernext/gpl-linux/acl.c, mmfs, avs_rcs3b, rcs3bs003a 4/13/05 12:59:46 */
 
-#if LINUX_KERNEL_VERSION > 2060000
-#ifdef CONFIG_FS_POSIX_ACL
+#if defined(XATTR_LINUX24_PATCH) || (LINUX_KERNEL_VERSION > 2060000)
 #include <Shark-gpl.h>
 
 #include <linux/string.h>
@@ -406,5 +405,4 @@
   EXIT(0);
   return rc;
 }
-#endif /* CONFIG_FS_POSIX_ACL */
-#endif /* LINUX_KERNEL_VERSION > 2060000 */
+#endif /* defined(XATTR_LINUX24_PATCH) || (LINUX_KERNEL_VERSION > 2060000) */
diff -bur src.orig/gpl-linux/inode.c src/gpl-linux/inode.c
--- src.orig/gpl-linux/inode.c	2005-04-29 01:35:18.000000000 +0200
+++ src/gpl-linux/inode.c	2006-04-05 14:02:21.000000000 +0200
@@ -1703,7 +1703,7 @@
   return rc;
 }
 
-#if LINUX_KERNEL_VERSION > 2060000
+#if defined(XATTR_LINUX24_PATCH) || (LINUX_KERNEL_VERSION > 2060000)
 #include <cxiAclUser.h>
 
 #define XATTR_SECURITY_PREFIX "security."
@@ -2160,4 +2160,4 @@
   EXIT(0);
   return (-rc);
 }
-#endif
+#endif /* defined(XATTR_LINUX24_PATCH) || (LINUX_KERNEL_VERSION > 2060000) */
diff -bur src.orig/gpl-linux/linux2gpfs.h src/gpl-linux/linux2gpfs.h
--- src.orig/gpl-linux/linux2gpfs.h	2005-04-29 01:35:19.000000000 +0200
+++ src/gpl-linux/linux2gpfs.h	2006-04-05 14:01:03.000000000 +0200
@@ -132,14 +132,14 @@
   int gpfs_i_getattr(struct dentry *, struct iattr *);
 #endif
 
-#if LINUX_KERNEL_VERSION > 2060000
+#if defined(XATTR_LINUX24_PATCH) || (LINUX_KERNEL_VERSION > 2060000)
   int gpfs_i_setxattr(struct dentry *, const char *, const void *, size_t, int);
   ssize_t gpfs_i_getxattr(struct dentry *, const char *, void *, size_t);
   ssize_t gpfs_i_listxattr(struct dentry *, char *, size_t);
   int gpfs_i_removexattr(struct dentry *, const char *);
   int gpfs_set_posix_acl(struct dentry *, int, const void *, size_t);
   int gpfs_get_posix_acl(struct dentry *, int, const void *, size_t);
-#endif
+#endif /* defined(XATTR_LINUX24_PATCH) || (LINUX_KERNEL_VERSION > 2060000) */
 
   int gpfs_i_getattr_internal(struct inode *iP);
   int gpfs_i_setattr_internal(struct inode *iP, struct iattr *aP);
diff -bur src.orig/gpl-linux/opsdeclare.c src/gpl-linux/opsdeclare.c
--- src.orig/gpl-linux/opsdeclare.c	2005-04-29 01:35:19.000000000 +0200
+++ src/gpl-linux/opsdeclare.c	2006-04-05 14:02:21.000000000 +0200
@@ -123,7 +123,7 @@
 #endif
   setattr:     gpfs_i_setattr,
   getattr:     gpfs_i_getattr
-#if LINUX_KERNEL_VERSION > 2060000
+#if defined(XATTR_LINUX24_PATCH) || (LINUX_KERNEL_VERSION > 2060000)
   ,
   setxattr: gpfs_i_setxattr,
   getxattr: gpfs_i_getxattr,
@@ -155,7 +155,7 @@
 #endif
   setattr:     gpfs_i_setattr,
   getattr:     gpfs_i_getattr
-#if LINUX_KERNEL_VERSION > 2060000
+#if defined(XATTR_LINUX24_PATCH) || (LINUX_KERNEL_VERSION > 2060000)
   ,
   setxattr: gpfs_i_setxattr,
   getxattr: gpfs_i_getxattr,
@@ -186,7 +186,7 @@
 #endif
   setattr:     gpfs_i_setattr,
   getattr:     gpfs_i_getattr
-#if LINUX_KERNEL_VERSION > 2060000
+#if defined(XATTR_LINUX24_PATCH) || (LINUX_KERNEL_VERSION > 2060000)
   ,
   setxattr: gpfs_i_setxattr,
   getxattr: gpfs_i_getxattr,
@@ -217,7 +217,7 @@
 #endif
   setattr:     gpfs_i_setattr,
   getattr:     gpfs_i_getattr
-#if LINUX_KERNEL_VERSION > 2060000
+#if defined(XATTR_LINUX24_PATCH) || (LINUX_KERNEL_VERSION > 2060000)
   ,
   setxattr: gpfs_i_setxattr,
   getxattr: gpfs_i_getxattr,
@@ -247,7 +247,7 @@
 #endif
   setattr:      gpfs_i_setattr,    /* setattr */
   getattr:      gpfs_i_getattr     /* getattr */
-#if LINUX_KERNEL_VERSION > 2060000
+#if defined(XATTR_LINUX24_PATCH) || (LINUX_KERNEL_VERSION > 2060000)
   ,
   setxattr: gpfs_i_setxattr,
   getxattr: gpfs_i_getxattr,
